docs/pallas/tpu/matmul.md

```diff
@@ -0,0 +1,643 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.0
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
++++ {"id": "-z6pOJwvn-_j"}
+
+# Pallas TPU - Your First Matmul
+
+In this guide, we'll write a matrix multiplication routine using Pallas. We'll also go over how to think about matmul performance on TPU and how to template a matmul kernel to fuse in operations.
+
+```{code-cell} ipython3
+:id: ejAVO6ikUUuF
+
+#@title Imports
+import functools
+from typing import Callable
+
+import jax
+from jax.experimental import pallas as pl
+from jax.experimental.pallas import tpu as pltpu
+from jax import random
+import jax.numpy as jnp
+import numpy as np
+```
+
++++ {"id": "58plJlycoPmT"}
+
+## Background
+
+Matrix multiplication is a fundamental linear algebra operation at heart of modern deep learning and language modeling. We'd like to make matmuls as speedy as possible using specialized accelerators like TPUs and GPUs. Both TPUs and GPUs have specialized units for fast matrix multiplication though they work differently.
```

Comment:
Remove "though they work differently"? Not sure what point you're trying to make here and you never expand on it.